### PR TITLE
Qickfix: Move Sandbox::checkSandboxed()

### PR DESCRIPTION
**Pull request by daschuer**
Move Sandbox::checkSandboxed() back to MixxxMainWindow fixing a crash introduced by #3956

This is a band aid. That reintroduces the issue of settings path access before checking the sandbox will be fixed in a follow up PR

### CODE REVIEW

**Comment by uklotzde** [comment]
Thanks. How about enforcing a restart after migrating the settings instead of trying to sync them afterwards? Starting over might be easier and safer.

**Comment by daschuer** [comment]
Yes that sounds be reasonable. This is the quick fix, merge?

**Comment by uklotzde** [comment]
> Yes that sounds be reasonable. This is the quick fix, merge?

But not before the build failures are fixed ;)

**Comment by daschuer** [comment]
Should work now. All this Sandbox::migrateOldSettings() is  big hack due to the convenience feature for developers to run a non sandboxed version in the sandboxed path. 

I think the issue from https://github.com/mixxxdj/mixxx/pull/3956 is still fixed here except of very early access to the preferences directory. This issue only applies when running as developer or during migration. 

If a developer noticed it, he can fix it. 

I am done here.

**Comment by daschuer** [comment]
Just for the record:
* The migration runs without translation
* There is no user feedback
* Logging does not go to Mixxx.log
* there is no second try in case of failures.

Solution:
* Run migration later after Mixxx is initialized than restart Mixxx.
* Separate the developers hack.

File: src/mixxx.cpp [hunk 1]
```diff
@@ -171,6 +171,13 @@ MixxxMainWindow::MixxxMainWindow(QApplication* pApp, const CmdlineArgs& args)
     m_runtime_timer.start();
     mixxx::Time::start();
 
+#ifdef __APPLE__
+    Sandbox::checkSandboxed();
```
Comment (uklotzde) [inline src/mixxx.cpp:175]: Please add a comment about the restrictions that have been violated before, i.e. **after** creating the application.

**Comment by daschuer** [comment]
OK comments improved. Maybe a Mac contributor picks is up one day.

**Comment by uklotzde** [comment]
@daschuer I fixed the typos directly and reordered/reworded the comment slightly. It was difficult to grasp even with my a priori knowledge about the context.

**Comment by daschuer** [comment]
Thank you.

**Review by uklotzde — APPROVED** [review]
Thank you! LGTM

**Pull request merged**